core/src/main/scala/com/avast/clients/rabbitmq/DefaultRabbitMQClientFactory.scala

```diff
@@ -118,38 +118,42 @@ private[rabbitmq] object DefaultRabbitMQClientFactory extends LazyLogging {
+    def fromConfig[F[_]: ConcurrentEffect, A: ProductConverter](
+        providedConfig: Config,
+        configName: String,
+        channel: ServerChannel,
+        factoryInfo: RabbitMQConnectionInfo,
+        blocker: Blocker,
+        monitor: Monitor)(implicit cs: ContextShift[F]): DefaultRabbitMQProducer[F, A] = {
       val producerConfig = providedConfig.wrapped(configName).as[ProducerConfig](configName)
-      create[F, A](producerConfig, configName, channel, factoryInfo, blockingScheduler, monitor)
+      create[F, A](producerConfig, configName, channel, factoryInfo, blocker, monitor)
     }
 
-    def create[F[_]: Effect, A: ProductConverter](producerConfig: ProducerConfig,
-                                                  configName: String,
-                                                  channel: ServerChannel,
-                                                  factoryInfo: RabbitMQConnectionInfo,
-                                                  blockingScheduler: Scheduler,
-                                                  monitor: Monitor)(implicit scheduler: Scheduler): DefaultRabbitMQProducer[F, A] = {
-      prepareProducer[F, A](producerConfig, configName, channel, factoryInfo, blockingScheduler, monitor)
+    def create[F[_]: ConcurrentEffect, A: ProductConverter](
+        producerConfig: ProducerConfig,
+        configName: String,
+        channel: ServerChannel,
+        factoryInfo: RabbitMQConnectionInfo,
+        blocker: Blocker,
+        monitor: Monitor)(implicit cs: ContextShift[F]): DefaultRabbitMQProducer[F, A] = {
+      prepareProducer[F, A](producerConfig, configName, channel, factoryInfo, blocker, monitor)
     }
 
   }
 
   object Consumer {
 
-    def fromConfig[F[_]: Effect, A: DeliveryConverter](
-        providedConfig: Config,
-        configName: String,
-        channel: ServerChannel,
-        channelFactoryInfo: RabbitMQConnectionInfo,
-        blockingScheduler: Scheduler,
-        monitor: Monitor,
-        consumerListener: ConsumerListener,
-        readAction: DeliveryReadAction[F, A])(implicit scheduler: Scheduler): DefaultRabbitMQConsumer[F] = {
+    def fromConfig[F[_]: ConcurrentEffect, A: DeliveryConverter](providedConfig: Config,
+                                                                 configName: String,
+                                                                 channel: ServerChannel,
+                                                                 channelFactoryInfo: RabbitMQConnectionInfo,
+                                                                 blocker: Blocker,
+                                                                 monitor: Monitor,
+                                                                 consumerListener: ConsumerListener,
+                                                                 readAction: DeliveryReadAction[F, A])(
+        implicit ec: ExecutionContext,
```

Comment:
Why is `ExecutionContext` needed? Should it be?

Reply:
There is a piece of code using `Future`s. I can try once again to get rid of it but didn't succeed so far.

Reply:
I think you should. Or at least it should be hidden from the user because it looks very strange that on one hand you're using `F` and on the other hand you require implicit `ExecutionCotext` which is usual for `Future`. At least I wouldn't pass it implicitly and I would take it as regular argument because it's necessary for the implementation.